File: Examples/VisualBasic/Mail-Merge/SimpleMailMerge.vb
Imports Aspose.Words
Imports Aspose.Words.MailMerging
Imports System.Collections.Generic
Imports System.Web

Class SimpleMailMerge
    Public Shared Sub Run()
        ' ExStart:SimpleMailMerge
        Dim Response As HttpResponse = Nothing
        ' The path to the documents directory.
        Dim dataDir As String = RunExamples.GetDataDir_MailMergeAndReporting()
        ' Open an existing document.
        Dim doc As New Document(dataDir & Convert.ToString("MailMerge.ExecuteArray.doc"))

        doc.MailMerge.UseNonMergeFields = True

        ' Fill the fields in the document with user data.
        doc.MailMerge.Execute(New String() {"FullName", "Company", "Address", "Address2", "City"}, New Object() {"James Bond", "MI5 Headquarters", "Milbank", "", "London"})

        dataDir = dataDir & Convert.ToString("MailMerge.ExecuteArray_out_.doc")
        ' Send the document in Word format to the client browser with an option to save to disk or open inside the current browser.
        doc.Save(Response, dataDir, ContentDisposition.Inline, Nothing)
        ' ExEnd:SimpleMailMerge
        Console.WriteLine(Convert.ToString(vbLf & "Simple Mail merge performed with array data successfully." & vbLf & "File saved at ") & dataDir)
    End Sub
End Class

File: Examples/CSharp/Mail-Merge/SimpleMailMerge.cs
using Aspose.Words;
using Aspose.Words.MailMerging;
using System;
using System.Collections.Generic;
using System.Data;
using System.Data.OleDb;
using System.Diagnostics;
using System.IO;
using System.Linq;
using System.Text;

namespace Aspose.Words.Examples.CSharp.Mail_Merge
{
    class SimpleMailMerge
    {
        // ExStart:SimpleMailMergeExecuteArray 
        public static void SimpleMailMergeExecuteArray()
        {
            // The path to the documents directory.
            string dataDir = RunExamples.GetDataDir_MailMergeAndReporting(); 

            // Include the code for our template.
            Document doc = new Document();
            DocumentBuilder builder = new DocumentBuilder(doc);

            // Create Merge Fields.
            builder.InsertField(" MERGEFIELD CustomerName ");
            builder.InsertParagraph();
            builder.InsertField(" MERGEFIELD Item ");
            builder.InsertParagraph();
            builder.InsertField(" MERGEFIELD Quantity ");

            builder.Document.Save(dataDir + "MailMerge.TestTemplate.docx");
            
            // Fill the fields in the document with user data.
            doc.MailMerge.Execute(new string[] { "CustomerName", "Item", "Quantity" },
                new object[] { "John Doe", "Hawaiian", "2" });

            builder.Document.Save(dataDir + "MailMerge.Simple.docx");
            // ExEnd:SimpleMailMergeExecuteArray
            Console.WriteLine("\nSimple Mail merge performed with array data successfully.\nFile saved at " + dataDir);
        }
    }
}
